lib/miq_expression.rb

```diff
@@ -309,8 +309,9 @@ def preprocess_for_sql(exp, attrs = nil)
       preprocess_for_sql(exp[operator], attrs)
       exp.delete(operator) if exp[operator].empty? # Clean out empty operands
     else
-      # check operands to see if they can be represented in sql
-      unless sql_supports_atom?(exp)
+      if sql_supports_atom?(exp)
+        convert_bytes_untits(exp) if %w[= != <= >= >].include?(operator)
```

Comment:
Small typo on method name. Should be `convert_bytes_units`

Reply:
@gtanzillo  renamed to slightly more descriptive - ```convert_size_in_units_to_integer```